### PR TITLE
Iterators and generators homework

### PR DESCRIPTION
**Pull request by katerynarieznik**

### CODE REVIEW

File: homework_iterators/fizzBuzz.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  if (!(n % 3)) return 'Fizz';
+  if (!(n % 5)) return 'Buzz';
+  if (!(n % 15)) return 'FizzBuzz';
+  return n;
+};
+
+//! Iterator solution
+
+const arr = [...Array(100).keys()].map((i) => i + 1);
+const arrIter = arr[Symbol.iterator]();
```
Comment (sharkich) [inline homework_iterators/fizzBuzz.js:11]: Do not use the internal iterator of an array - you need to create a new one by your self.

Comment (sharkich) [reply homework_iterators/fizzBuzz.js:11]: remove `arr`. use a pure loop instead.

File: homework_iterators/randomNumbers.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+const MAX = 1000;
+const QUANTITY = 10;
+
+const randomNumber = generateRandomNumbers(MAX, QUANTITY);
+
+for (let i = 0; i < QUANTITY; i++) {
+  console.log(randomNumber.next().value);
+}
```
Comment (sharkich) [inline homework_iterators/randomNumbers.js:18]: use `for of`

**Review by sharkich — CHANGES_REQUESTED** [review]

File: homework_iterators/fizzBuzz.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    yield checkFizzBuzz(i);
+    i++;
+  }
+  return checkFizzBuzz(i);
+}
+
+const fizzBuzz = fizzBuzzGenerator();
+
+// Print
+for (let i = 0; i < 100; i++) {
```
Comment (sharkich) [inline homework_iterators/fizzBuzz.js:50]: use `for of` instead, because this is duplicate logic from the iterator (limit of 100)

**Review by sharkich — CHANGES_REQUESTED** [review]

File: homework_iterators/fizzBuzz.js (hunk 3 above)
Comment (rieznik) [reply homework_iterators/fizzBuzz.js]: Oh, these iterators and generators are a bit confusing for me 🥲

**Pull request merged**